### PR TITLE
Streamlined hsctl nodejs deploy

### PR DESCRIPTION
**Pull request by dan-palmer1**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev
These steps are for the dev-hs-x instances:
     1)  Log onto a dev-hs-x instance: ssh [user]@dev-hs-x
          a. Change to hydro-service user: sudo -u hydro-service -s
          b. cd ~/hydroshare
          c. ./hsctl rebuild --db

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation
  Runs as normal ./hsctl rebuild --db run.
  Code is part of hsctl.

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW BUGS | 15 |
| NEW DUPLICATED LINES DENSITY | 14.469994391475 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW VULNERABILITIES | 0 |
| NEW CODE SMELLS | 9 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3973](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3973)

File: config/hydroshare-config.yaml [hunk 1]
```diff
@@ -1,57 +1,49 @@
 ### Local Configuration Variables ###
 HS_PATH: /home/hydro/hydroshare
-HOST_SSL_DIR: /home/hydro/hs-certs
-HS_DATABASE: pg.development.sql
+HOST_SSL_DIR: /home/hydro/hydroshare/nginx/cert-files
```
Comment (sblack-usu) [inline config/hydroshare-config.yaml:3]: I don't understand why there are changes in this file

File: hsctl [hunk 2]
```diff
@@ -419,10 +423,48 @@ reset_all_hs() {
...
+cd static/
+mv js/app.*.js js/app.js
+mv js/chunk-vendors.*.js js/chunk-vendors.js
+cd ..
+eof
+
+docker container rm nodejs
+sleep 1
+}
+
```
Comment (sblack-usu) [inline hsctl:463]: I thought there had to be a permissions change on the folder here?

File: hsctl [hunk 3]
```diff
@@ -369,6 +372,7 @@ rebuild_hs() {
     rebuild_solr_index
     sleep 2s
     restart_hs
+    sleep 2s
```
Comment (sblack-usu) [inline hsctl:375]: All of these sleeps... are any of them necessary?

File: hsctl (hunk 2 above)
Comment (dan-palmer1) [reply hsctl:463]: Because of the placement of the call, the permission changes were already being done in a later step of the rebuild.

File: hsctl (hunk 3 above)
Comment (dan-palmer1) [reply hsctl:375]: Those have always been there.  I did not add any extra sleeps to hsctl.

File: hydroshare/urls.py [hunk 4]
```diff
@@ -143,7 +143,7 @@
     # one out.
 
     # url("^$", direct_to_template, {"template": "index.html"}, name="home"),
-    url(r"^tests/$", direct_to_template, {"template": "tests.html"}, name="tests"),
+    # QUNIT_TESTS_OFF
```
[the remark this answers is not in the document]
Comment (dan-palmer1) [reply hydroshare/urls.py:147]: This file was not intended to be part of this PR.  The changes shown are done separately by Hydroshare when rebuilt. 
I added no new changes to this file.

File: config/hydroshare-config.yaml (hunk 1 above)
Comment (dan-palmer1) [reply config/hydroshare-config.yaml:3]: This file was not intended to be part of this PR. The changes shown are done separately by Hydroshare when rebuilt.
I added no new changes to this file.

Comment (sblack-usu) [reply config/hydroshare-config.yaml:3]: The changes were merged, we'll have to back them out then.

File: hsctl (hunk 3 above)
Comment (sblack-usu) [reply hsctl:375]: There are like 10 seconds of sleep added in this diff.

File: hsctl [hunk 5]
```diff
@@ -419,10 +423,48 @@ reset_all_hs() {
...
+
+docker container rm nodejs
+sleep 1
+}
+
 ### Display usage if exactly one argument is not provided ###
 if [  $# -ne 1 ]
 then
-    if [ $1 != "rebuild" ] || [ $2 != "--db" ] && [ $1 != "managepy" ]
+    if [ $1 != "rebuild" ] && [ $2 != "--db" ] && [ $1 != "managepy" ]
```
Comment (sblack-usu) [inline hsctl:467]: Why was this changed?
